lib/remote/apilistener.cpp

```diff
@@ -1575,14 +1579,8 @@ void ApiListener::ReplayLog(const JsonRpcConnection::Ptr& client)
 		}
 
 		if (last_sync) {
-			{
-				ObjectLock olock2(endpoint);
-				endpoint->SetSyncing(false);
-			}
-
 			OpenLogFile();
-
-			break;
+			return;
 		}
```

Comment:
Now `OpenLogFile()` won't be called if the `last_sync` iteration return early.

I see two options to address this:
1. Use `Defer`, capture `last_sync` and only do it conditionally (the quick fix).
2. Change `Defer` so that it can be initialized empty (to define it's scope) and be (conditionally) set to a function later (a little bigger change, but the code here would probably look nicer).

Reply:
Either way, this will not work reliably since neither of those have the ability to open the log file while holding the lock. So, I'm going to revert the latest changes.

Reply:
> Either way, this will not work reliably since neither of those have the ability to open the log file while holding the lock.

Why is that the case?

Reply:
The mutex is locked within the for loop, how is a deferred callback going to open the log file while this lock is still in scope?

Reply:
```c++
for (;;) {
	std::unique_lock<std::mutex> lock(m_LogLock);

	CloseLogFile();
	Defer reopenLog;

	if (count == -1 || count > 50000) {
		OpenLogFile();
		lock.unlock();
	} else {
		reopenLog.Set([this]() { OpenLogFile(); });
		last_sync = true;
	}

	// ...
}
```

Reply:
At first I thought you meant that the defer should be initiated within the function scope and not within the loop.